File: 2025/07/07_day.py

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import joypy
import seaborn as


# Load data --------

pokemon_df = pd.read_csv('https://raw.githubusercontent.com/rfordatascience/tidytuesday/main/data/2025/2025-04-01/pokemon_df.csv')


# clean data ------

# font family
plt.rcParams["font.family"] = "Candara"

pokemon_df.columns



# Select relevant columns
stats = ['attack', 'defense', 'speed']
selected_types = ['fire', 'water', 'grass', 'electric']
col = ['#78C850', '#C03028', '#6890F0', '#f4cd2c']


df = pokemon_df[['pokemon', 'type_1'] + stats]
df = df[df['type_1'].isin(selected_types)].dropna()

# Long format
df_long = df.melt(
    id_vars=['pokemon', 'type_1'],
    value_vars=stats,
    var_name='stat',
    value_name='value'
)

# Create FacetGrid
g = sns.FacetGrid(
    df_long,
    col='type_1',
    col_order=selected_types,
    sharey=True,
    height=4,
    aspect=0.8
)

# Loop over axes and draw violin + points manually
for ax, t, c in zip(g.axes.flat, selected_types, col):
    subset = df_long[df_long['type_1'] == t]
    
    sns.violinplot(
        data =subset,
        x = 'stat',
        y = 'value',
        inner = None,
        linewidth = 0.8,
        edgecolor = 'black',
        bw = 0.3,
        color = c,
        ax = ax, 
        alpha = 0.4
    )
    
    sns.stripplot(
        data = subset,
        x = 'stat',
        y = 'value',
        color = c,
        size = 5,
        jitter = .15,
        alpha = 0.9,
        ax = ax,
        edgecolor = 'white',  # This sets the edgecolor to white
        linewidth = 0.25        # This controls the thickness of the edge
    
    )
    
    ax.set_title(t.capitalize())
    ax.set_xlabel("")
    ax.set_ylabel("")

# Final layout
plt.suptitle("Distribution of Pokémon Stats by Type", fontsize=16)
plt.tight_layout()
plt.show()






# ax.hlines(
    # goal_comparison["Team"], 
    # 0, 
    # goal_comparison["Goals Scored"], 
    # color = "#A6ACAD",
    # alpha = 0.6
# )
# 
# ax.scatter(
    # goal_comparison["Goals Scored"], 
    # goal_comparison["Team"], 
    # color = "#5f899d", 
    # label = "Goals Scored", 
    # s = 80,
    # zorder = 3
# )
# 
Add text for goals scored
# for i, (team, goals) in enumerate(zip(goal_comparison["Team"], goal_comparison["Goals Scored"])):
    # ax.text(
        # goals + 4, 
        # team, 
        # str(goals), 
        # va = "center", 
        # ha = "left", 
        # fontweight = "bold",
        # fontsize = 9, 
        # color = "#5f899d"
    # )
# 
Lollipops for goals conceded
# ax.hlines(
    # goal_comparison["Team"], 
    # goal_comparison["Goals Conceded"], 
    # 0, 
    # color = "#A6ACAD", 
    # alpha = 0.6
# )
# 
# ax.scatter(
    # goal_comparison["Goals Conceded"], 
    # goal_comparison["Team"], 
    # color = "#dc756e", 
    # label="Goals Conceded", 
    # s = 80,
    # zorder = 3
# )
# 
Add text for goals conceded
# for i, (team, goals) in enumerate(zip(goal_comparison["Team"], goal_comparison["Goals Conceded"])):
    # ax.text(
        # goals - 4, 
        # team, 
        # str(abs(goals)), 
        # va = "center", 
        # ha = "right", 
        # fontweight = "bold",
        # fontsize = 9, 
        # color = "#dc756e"
    # )
# 
Add a vertical line at 0
# ax.axvline(0, color = "black", linewidth = 1 , linestyle="dashed")
# 
# 
Update x-axis labels to show absolute values (convert negatives to positives)
# ax.set_xticks(ax.get_xticks())  # Keep the original tick positions
# ax.set_xticklabels([f"{int(abs(tick))}" for tick in ax.get_xticks()])
# 
Customization
# ax.set_xlabel("Goals")
# 
legend
# handles, labels = ax.get_legend_handles_labels()
# ax.legend(handles[::-1], labels[::-1], loc = 'center left', bbox_to_anchor=(1.02, 0.5), fontsize = 8)  
# 
# plt.title(
    # "Premier League: Goals Conceded vs Goals Scored (Season 2021-2022). ",
    # fontsize = 12,
    # fontweight = "bold",
    # pad = 35,
    # x = 0.5,
    # y = 0.99
# )
# 
Add the subtitle for clarification
# plt.text(x = 0.5, y = 1.05, 
        # s =  "The further a team’s points extend from the center,\n"
        #    "the more dominant or vulnerable they were in attack or defense.",
        # ha = 'center', 
        # va = 'center', 
        # fontsize = 10, 
        # style = 'italic', 
        # color = "#8C8380",
        # transform = plt.gca().transAxes
# )
# 
# 
Add a caption to the plot
# plt.text(
    # x = 0.98, y = -0.15,  # Adjust x, y to position the
    # s = "Source:  Premier League Match Data 2021-2022 | Graphic: Natasa Anastasiadou",
    # ha = 'center', 
    # va = 'center', 
    # fontsize = 6,
    # fontweight = "bold",  
    # style = 'italic', 
    # color = "#8C8380",
    # transform = plt.gca().transAxes
# )
# 
# 
# 
Add gridlines only for the x-axis
# ax.grid(axis="x", linestyle="--", alpha=0.2)
# 
# 
Display the plot
# plt.tight_layout()
# plt.show()
# 
# 
# plt.savefig("plot.png", dpi = 600, bbox_inches='tight')
# 

File: 2025/07/07_day.R


rm(list = ls())
gc()


# load libraries -------

library(data.table)
library(ggplot2)
library(stringr)
library(ggtext)
library(extrafont)
library(paletteer)
library(tidyverse)


# load data --------

office_ratings <- fread('https://raw.githubusercontent.com/rfordatascience/tidytuesday/main/data/2020/2020-03-17/office_ratings.csv')


office_ratings$season <- office_ratings$season |> as.character()



# outliers 

outliers <- office_ratings[imdb_rating > 9.5 | imdb_rating <= 7]


office_ratings_clean <- office_ratings[!(imdb_rating > 9.5 | imdb_rating <= 7)]



colors = paletteer_c("ggthemes::Sunset-Sunrise Diverging", 9)

# Plot 1 -----------

gr <- ggplot() +
    
    geom_boxplot(
        data = office_ratings_clean,
        aes(x = season, y = imdb_rating, fill = season),
        position = position_dodge(width = .5),
        width = .35, outlier.shape = NA, alpha = 0.45, linewidth = 0.2
    ) +
    
    geom_point(
        data = office_ratings_clean,
        aes(x = season, y = imdb_rating, fill = season),
        position = position_jitterdodge(jitter.width = .25, dodge.width = .5),
        shape = 21, size = 2.5, stroke = .15, color = "white"
    ) +
    
    geom_point(
        data = outliers,
        aes(x = season, y = imdb_rating, fill = season),
        shape = 21, size = 4.5, stroke = 0.25, color = "white"
    ) +
    # 
    # geom_text(
    #     data = outliers,
    #     aes(x = season, y = imdb_rating, label = paste0("Ep. ", episode)),
    #     # position = position_jitter(width = 0.25, height = 0),
    #     family = "Candara",
    #     size = 3.5,
    #     hjust = -0.4,
    #     vjust = 0.4,
    #     check_overlap = TRUE
    # ) +
    # 
    
    geom_label(
        data = outliers,
        aes(x = season, y = imdb_rating, label = paste0("Ep. ", episode)),
        # position = position_jitter(width = 0.25, height = 0),
        family = "Candara",
        size = 3.5,
        label.size = 0.2,     # border thickness
        label.padding = unit(0.15, "lines"),
        label.r = unit(0.15, "lines"),  # border radius
        color = "grey35",
        hjust = -0.4,
        alpha = 0.15
    ) +
    
    scale_fill_manual(values = colors) +
    
    scale_color_manual(values = colors) +
    
    theme_minimal() +
    
    labs(
        title = "The Office: IMDb Ratings Across Seasons",
        subtitle = "Rating distribution per season. Outliers (above 9.5 or below and equal 7) are labeled.",
        caption = "Source: <b>The Office ratings</b> | Graphic: <b>Natasa Anastasiadou</b>",
        x = "Season",
        y = "IMDb Rating"
    ) +
    
    theme(
        legend.position = "none",
        
        axis.title.x = element_text(size = 10, family = "Candara"),
        axis.title.y = element_text(size = 10, family = "Candara"),
        axis.text.x = element_text(size = 10, family = "Candara"),
        axis.text.y = element_text(size = 10, family = "Candara"),
        
        panel.grid.major = element_line(linewidth = .35, color = "grey80", linetype = "dashed"),
        panel.grid.minor = element_line(linewidth = .35, color = "grey80", linetype = "dashed"),
        
        plot.title = element_markdown(size = 16, face = "bold", hjust = 0.5, family = "Candara", margin = margin(t = 15, b = 5)),
        plot.subtitle = element_markdown(size = 11, hjust = 0.5, family = "Candara", color = "grey30", margin = margin(t = 5, b = 25)),
        plot.caption = element_markdown(margin = margin(t = 20), size = 8, family = "Candara", hjust = 1),
        
        plot.margin = margin(20, 20, 20, 20),
        
        plot.background = element_rect(fill = "#e4e4e3", color = NA)
    )

gr

# save ---------

ggsave(
   plot = gr, filename = "Rplot.png",
   width = 10, height = 8, units = "in", dpi = 600
)

